### ISSUE TITLE
SQL Server: Column length missmatch between locales.code and *.localeCode when using a DB generated with wiki.js pre-2.0.0-beta.241

### ISSUE THREAD

**Issue by AchimHentschel** [Requarks/wiki#1152]
# Describe the bug
I just upgraded my test environment to Wiki.js 2.0.0-rc.17. Again there is an issue with the SQL Server DB backend (maybe other backends might end with the same error though). Cause of this issue is commit https://github.com/Requarks/wiki/commit/08fd10603f7fdfdbe98bffe37e92630f11db2f91 which was introduced during version 2.0.0-beta.230 and 2.0.0-beta.241. This commit hard-changed the column size of the locales.code column from 2 to 5. Since this change was not introduced as a migration recreation of the pageTree table fails in migration 2.0.0-rc.2.js when the foreign key constraint between pageTree and locales will be recreated. In SQL Server both columns, locales.code (length 2 in my pre-beta.241 database) and pageTree.localeCode (length 5, due to re-creation) need to have the same length and scale.

You can fix this startup behaviour by executing the following T-SQL script:
```sql
ALTER TABLE [dbo].[users] DROP CONSTRAINT [users_localecode_foreign]
GO

ALTER TABLE [dbo].[pages] DROP CONSTRAINT [pages_localecode_foreign]
GO

ALTER TABLE [dbo].[pageHistory] DROP CONSTRAINT [pagehistory_localecode_foreign]
GO

ALTER TABLE [dbo].[pageTree] DROP CONSTRAINT [pagetree_localecode_foreign]
GO

ALTER TABLE locales ALTER COLUMN code nvarchar(5) not null
ALTER TABLE users ALTER COLUMN localeCode nvarchar(5) not null
ALTER TABLE pages ALTER COLUMN localeCode nvarchar(5) not null
ALTER TABLE pageHistory ALTER COLUMN localeCode nvarchar(5) not null
ALTER TABLE pageTree ALTER COLUMN localeCode nvarchar(5) not null
GO

ALTER TABLE [dbo].[users]  WITH CHECK ADD  CONSTRAINT [users_localecode_foreign] FOREIGN KEY([localeCode])
REFERENCES [dbo].[locales] ([code])
GO

ALTER TABLE [dbo].[users] CHECK CONSTRAINT [users_localecode_foreign]
GO

ALTER TABLE [dbo].[pages]  WITH CHECK ADD  CONSTRAINT [pages_localecode_foreign] FOREIGN KEY([localeCode])
REFERENCES [dbo].[locales] ([code])
GO

ALTER TABLE [dbo].[pages] CHECK CONSTRAINT [pages_localecode_foreign]
GO

ALTER TABLE [dbo].[pageHistory]  WITH CHECK ADD  CONSTRAINT [pagehistory_localecode_foreign] FOREIGN KEY([localeCode])
REFERENCES [dbo].[locales] ([code])
GO

ALTER TABLE [dbo].[pageHistory] CHECK CONSTRAINT [pagehistory_localecode_foreign]
GO

ALTER TABLE [dbo].[pageTree]  WITH CHECK ADD  CONSTRAINT [pagetree_localecode_foreign] FOREIGN KEY([localeCode])
REFERENCES [dbo].[locales] ([code])
GO

ALTER TABLE [dbo].[pageTree] CHECK CONSTRAINT [pagetree_localecode_foreign]
GO
```

# To Reproduce
1. Steps to reproduce the behavior:
Install wiki.js version 2.0.0-beta.230 and let it create a fresh SQL Server DB with locales.code length of 2.
2. Stop beta.230
3. Startup the new Wiki.js 2.0.0-rc.17 build, which introduced the first migration with relation to the locales.code column.

# Expected behavior
There should have been a clean migration for column size 2 to 5 - not just a change in an existing migration. The expectation was a clean startup.

# Host Info (please complete the following information):

OS: Debian Version 9.9, docker-ce 19.03.2
Wiki.js version: Wiki.js 2.0.0-beta.275
Database engine: mssql

**Comment by AchimHentschel** [comment]
A migration to the latest version fails. That should make it possible to either fix the 2.0.0-rc.2.js migration script or introduce a further migration, for example 2.0.0-beta.241. There should not be any databases out there (except for manually fixed DBs like mine) where migration 2.0.0-rc.2.js has been executed successfully. What do you think? I am willing to try a patch implementation for the column length. I am just not sure, which scenario would be best regarding this issue. My stomach would choose a new migration script 2.0.0-beta.241.js.

**Comment by AchimHentschel** [comment]
By the way. It was a bit harder to find the exact error message. In the logs I just saw
> 2019-10-28T08:48:53.191Z [MASTER] info: Database Connection Successful [ OK ]
> migration file "2.0.0-rc.2.js" failed
> migration failed with error: ALTER TABLE [pageTree] ADD CONSTRAINT [pagetree_localecode_foreign] FOREIGN KEY ([localeCode]) REFERENCES [locales] ([code]) - Could not create constraint or index. See previous errors.
> 2019-10-28T08:48:53.263Z [MASTER] error: Database Initialization Error: Transaction has been aborted.

Nothing logged about the column length issue - guess that is how knex works... I Used the XEvent Profiler implemented in SQL Server Management Studio to view the SQL statements issued to the DB. Then it was clear what exactly went wrong here.

**Comment by NGPixel** [comment]
Fixed by 8eddc4799e134dc568dbf942559df56d4bee8eb4
